parquet-thrift/src/main/java/org/apache/parquet/thrift/ThriftRecordConverter.java

```diff
@@ -445,6 +600,26 @@ public FieldEnumConverter(List<TProtocol> events, ThriftField field) {
       }
     }
 
+     @Override
+     protected ParquetProtocol decodeFromDictionary(Dictionary d, int i) {
+       Binary value = d.decodeToBinary(i);
+       final Integer id = enumLookup.get(value);
```

Comment:
This method and addBinary below. Actually, lots of methods are duplicating the inline class definition, which ends up creating two identical .class files that get optimized separately. Code reuse there would probably be a good thing.